### PR TITLE
`Shw::Flex` GTS Conversion

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR updates the `Shw::Flex` component to the new GTS format.

### :link: External links

Jira ticket: [HDS-3381](https://hashicorp.atlassian.net/browse/HDS-3831)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3381]: https://hashicorp.atlassian.net/browse/HDS-3381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: showcase/app/components/shw/flex/index.gts [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
 import Component from '@glimmer/component';
 import { htmlSafe } from '@ember/template';
+import { hash } from '@ember/helper';
+import Item from './item';
+import Label from '../label';
```
Comment (didoo) [inline showcase/app/components/shw/flex/index.gts:10]: [chore] better to preserve the namespacing, so when we look at the code we know where a component instance comes from (`<Label>` vs `<ShwLabel>`, `<Item>` vs `<ShwFlexItem>`)
```suggestion
import ShwFlexItem from './item';
import ShwLabel from '../label';
```

File: showcase/app/components/shw/flex/item.gts [hunk 2]
```diff
@@ -4,8 +4,10 @@
  */
 
 import Component from '@glimmer/component';
-import type { ComponentLike } from '@glint/template';
+import { hash } from '@ember/helper';
+import Label from '../label';
```
Comment (didoo) [inline showcase/app/components/shw/flex/item.gts:8]:
```suggestion
import ShwLabel from '../label';
```

File: showcase/app/components/shw/flex/item.gts [hunk 3]
```diff
@@ -4,8 +4,10 @@
...
 
+import type { ComponentLike } from '@glint/template';
```
Comment (didoo) [inline showcase/app/components/shw/flex/item.gts:10]: [question]: I don't see this used? 🤔

**Review by didoo — COMMENTED** [review]
left a few comments

File: showcase/app/components/shw/flex/item.gts (hunk 3 above)
Comment (zamoore) [reply showcase/app/components/shw/flex/item.gts:10]: It's used in the block for defining the type for the `Label` prop (L:21)

Comment (didoo) [reply showcase/app/components/shw/flex/item.gts:10]: 🤦‍♂️ (it's `.gts`, these things need to be explicitly imported)

File: showcase/app/components/shw/flex/index.gts [hunk 4]
```diff
@@ -5,6 +5,9 @@
 
 import Component from '@glimmer/component';
 import { htmlSafe } from '@ember/template';
+import { hash } from '@ember/helper';
+import ShwFlexItem from './item';
+import ShwFlexLabel from '../label';
```
Comment (didoo) [inline showcase/app/components/shw/flex/index.gts:10]: [chore] Sorry to bother you, maybe I did the wrong suggestion, but this is `ShwLabel` not `ShwFlexLabel`
```suggestion
import ShwLabel from '../label';
```

File: showcase/app/components/shw/flex/item.gts [hunk 5]
```diff
@@ -4,8 +4,10 @@
  */
 
 import Component from '@glimmer/component';
-import type { ComponentLike } from '@glint/template';
+import { hash } from '@ember/helper';
+import ShwFlexLabel from '../label';
```
Comment (didoo) [inline showcase/app/components/shw/flex/item.gts:8]:
```suggestion
import ShwLabel from '../label';
```

**Review by aklkv — APPROVED** [review]
Aside from `ShwFlexLabel` -> `ShwLabel` Looks good 👍🏻

**Pull request merged**